### DISCUSSION TITLE
btl2capfix module is resetting WearOS (Galaxy Watch 4) connection

### DISCUSSION THREAD

**Discussion by SriviharReddy** [kavishdevar/librepods#7] [Q&A] [answered]
I tried the module on both Android 14 & 15 stock ROMs, while ALN is working properly with the Airpods Pro 2, the connection to my Galaxy Watch 4 is getting reset and is forcing me setup the watch all over every ~15 minutes. This is happening on both android 14 & 15 and it stopped when I uninstalled the btl2capfix module.
Is there something I'm doing wrong?

**Comment by kavishdevar** [comment]
Thank you for trying the project out!

Your audio and the app works. Is that right?

**Comment by kavishdevar** [comment]
I also use a Galaxy Watch (6) with this library! I haven't been forced to reset yet. Could you share more details on what you mean by forcing you to reset? Maybe a few screenshots?

**Comment by SriviharReddy** [comment, marked as answer]
Thank you for the response. Looks like its a galaxy watch issue. The watch requires a reset everytime it's disconnected from the phone. It's very annoyying. 
https://eu.community.samsung.com/t5/wearables/galaxy-watch-4-keeps-asking-to-reset-and-re-pair/td-p/6767475

**Reply by kavishdevar** [reply]
Okay! But, please do share logs if you can, just in case it is a bug with the library. If you're not using a Samsung phone, they try to give you a hint of bad experience with watches. I can not register my eSIM without a samsung phone on my watch.

I’ve also had this issue sometimes where the google play services app on the watch it just keeps on asking me to repair. I'd suggest you to reset the watch before you get the notification, clear Google Play Services app’s data, and reinstall both the wearable app, and the plugin!

**Reply by SriviharReddy** [reply]
Yeah. That seems to have resolved it. 
Thank you for making the app. I got my Airpods Pro 2 as a gift and not being able to control their ANC had me very frustrated. I'm thankful that I came across your project.

**Reply by kavishdevar** [reply]
Thank you! Glad to hear my project was useful! :D

**Reply by kavishdevar** [reply]
Do share the project (and the bug report on google's issue tracker for upvotes) with people whom you know that use AirPods! 

fun story: I got airpods pro 2 for scoring good in my 10th board exam, and I've been developing this project ever since! I've spent more time developing, than using :)

**Reply by SriviharReddy** [reply]
Yeah. I upvoted the bug report.

haha. I tinker with my stuff a lot more than I use  them too.

**Comment by SriviharReddy** [comment]
![Screenshot_20241202-205936_Pixel Launcher](https://github.com/user-attachments/assets/e6c23e62-7568-4980-9b76-a821ec7f0d83)
I've had another minor issue. The QS tile is showing is not working and is only showing "unavailable" even when the Airpods are connected.

**Reply by kavishdevar** [reply]
oops, that's not supposed to be there yet! But look out for a new release! I'll be releasing a new version with automatic connection and many more settings soon!

**Reply by kavishdevar** [reply]
btw, if you rename using the app, you would have to repair with the phone, currently there is no way for a user app to rename a bluetooth device. Once you repair, your phone will show the new name!

**Reply by SriviharReddy** [reply]
Oh. I was wondering about the renaming option.

**Comment by kavishdevar** [comment]
Also, if you would like to test out in-development features, I'll be creating a channel on Telegram for future alpha releases and discussions on bugs! GitHub is of course there, but I'll be much more active over on Telegram (if I can spare time from JEE preparations, that is)

**Reply by SriviharReddy** [reply]
Sure.

**Comment by SriviharReddy** [comment]
I was a little confused by the instructions for the workaround in the readme, so I want to update them and the readme overall. is that okay?

**Reply by kavishdevar** [reply]
Yes! Contributions are welcome, thanks! Make a PR!

**Comment by SriviharReddy** [comment]
![Screenshot_20241203-133907_Settings](https://github.com/user-attachments/assets/d8596a5f-19a2-4d5c-a9ee-43ae55079d5d)
![Screenshot_20241203-133922_Bluetooth Devices Helper](https://github.com/user-attachments/assets/d3827581-f871-493b-8b68-996fdf07791c)
Did your app add the icons and battery to the bluetooth settings screen?

**Reply by kavishdevar** [reply]
nope! that'd be the app [BtHelper](https://github.com/TheParasiteProject/packages_apps_BtHelper/) that comes with many custom roms

**Reply by kavishdevar** [reply]
I've tried implementing my code into the app itself, but my PC is too slow to compile a system app...

**Reply by kavishdevar** [reply]
I am looking to make implement this in my app that does this, with root, someday, hopefully. So, if the user is rooted, it can do all this as well

**Reply by kavishdevar** [reply]
Hang on a second, SPATIAL AUDIO!? What else is in the settings? Which ROM is it?

**Reply by SriviharReddy** [reply]
> I am looking to make jmplement this in my app that does this, with root, someday, hopefully. So, if the user is rooted, it can do all this as well

Thats very cool.

> Hang on a second, SPATIAL AUDIO!? What else is in the settings? Which ROM is it?

https://youtu.be/EsTOVno1DEY?si=6XqLM_bgh5pDHlP7
I think spatial audio is a feature of the google pixel and is available on all pixel based roms. Can't find a way to test if it is working tho. Which app should I use?

**Reply by kavishdevar** [reply]
You could try apple music!

**Comment by kavishdevar** [comment]
@SriviharReddy do join https://t.me/airpodslikenormal if you'd like!
